screed/tests/test_attriberror.py

```diff
@@ -0,0 +1,90 @@
+import screed
+from screed.DBConstants import fileExtension
+import os
+import screed_tst_utils as utils
+import shutil
+
+
+class nostring:
+    def __init__(self):
+        self.exists = True
+
+    def __str__(self):
+        return ""
+
+    def __repr__(self):
+        return ""
+
+
+class test_comparisons():
+
+    def setup(self):
+        self._testfile = utils.get_temp_filename('test.fa')
+        shutil.copy(utils.get_test_data('test.fa'), self._testfile)
+        screed.read_fasta_sequences(self._testfile)
+
+        self._db = screed.ScreedDB(self._testfile)
+        self._ns = nostring()
+
+    def test_eq(self):
+        for k in self._db:
+            record = self._db.get(k)
+            res = (record.sequence == self._ns)
+            assert res == False, res
+
+    def test_neq(self):
+        for k in self._db:
+            record = self._db.get(k)
+            res = (record.sequence <> self._ns)
+            assert res == True, res
+
+    def test_comp_greateq(self):
+        for k in self._db:
+            record = self._db.get(k)
+            res = (record.sequence >= self._ns)
+            assert res == True, res
```

Comment:
I suspect what's happening here is that since the `sequence`'s __ge__ returns `NotImplemented` the interpreter failsover to the `__lt__` method of the `ns` and nots the result, and since a string (even an empty string) is "greater than" `NotImplemented` you end up with `True`